coc/events.py

```diff
@@ -765,6 +781,34 @@ def _task_callback_check(self, result):
             self._updater_tasks[name] = self.loop.create_task(lookup[name]())
             self._updater_tasks[name].add_done_callback(self._task_callback_check)
 
+    async def _raid_poller(self):
+        # pylint: disable=broad-except, protected-access
+        try:
+            age = 0
+            while self.loop.is_running():
+                try:
+                    [raidlog_entry] = await self.get_raidlog("#2PP", limit=1)
+                    raidlog_entry: coc.raid.RaidLogEntry
+                except Maintenance:
+                    await asyncio.sleep(15)
+                except Exception:
+                    await asyncio.sleep(DEFAULT_SLEEP)
+                else:
+                    if raidlog_entry.start_time.seconds_until + age > 0 and raidlog_entry.end_time.seconds_until > 0:
+                        # raid started
+                        self.dispatch("raid_weekend_start")
+                    elif raidlog_entry.end_time.seconds_until + age > 0 and 0 > raidlog_entry.end_time.seconds_until:
+                        # raid ended
+                        self.dispatch("raid_weekend_end")
+                    # sleep for response_retry + 1
+                    age = raidlog_entry._response_retry + 1
+                    await asyncio.sleep(age)
+        except asyncio.CancelledError:
```

Comment:
These should be their own function to be expressive. This looks cleaner:

```py
if raid_started(args):
    dispatch()
    ```

Reply:
where would you define such a function?

Reply:
i don't understand, can you clarify?

Reply:
I think what he wants is
```
def raid_started(raidlog_entry: coc.RaidlogEntry) -> bool:
    return raidlog_entry.start_time.seconds_until + age > 0 and raidlog_entry.end_time.seconds_until > 0
```
and then later
```
if raid_started(raidlog_entry):
    self.dispatch("raid_weekend_start")
```

Reply:
Since I this is a code style thing, I am not sure where to put that function definition.

Reply:
I’m this file is fine. Just make it private.